### ISSUE TITLE
Possible future works

### ISSUE THREAD

**Issue by benedekh** [FTSRG/BME-MODES3#85]
- [ ] implement new message formats, e.g. JSON, XML. As of now (04.09.2017) only protobuf is used.
- [ ] enable different message formats to be used on the network in parallel. E.g. JSON, protobuf messages are being transferred through the network and each of them is correctly handled by the dispatchers
- [x] separate the messages for different channels on the MQTT transport. As of now (04.09.2017) every message goes to the _/modes3/data_ channel, which might not be the best solution. **Note** that it might require changing the deployment scripts also. It looks like a big refactoring.
- [ ] any other issues which are marked as __wontfix__ on GitHub (first, check if they are still relevant)
- [ ] any other issues which are marked as __enhancement__ on GitHub (first, check if they are still relevant)

**Comment by benedekh** [comment]
> separate the messages for different channels on the MQTT transport. As of now (04.09.2017) every message goes to the /modes3/data channel, which might not be the best solution. Note that it might require changing the deployment scripts also. It looks like as a big refactoring.

is already implemented on the `refactoring` branch. --> *update*: it is merged into `master`
